test/run-tests

```diff
@@ -56,25 +67,33 @@ def handle_transient_httperrors(error):
 
     # Handle transient server-side errors
     if error.response.status_code in (500, 502, 503, 504):
-        global transient_error_wait_time
+        timeout = transient_error_wait_time.next()
         print(
             ">>> Server returned {} {}, waiting {} s...".format(
-                error.response.status_code,
-                error.response.reason,
-                transient_error_wait_time,
+                error.response.status_code, error.response.reason, timeout
             )
         )
-        time.sleep(transient_error_wait_time)
-        transient_error_wait_time = min(
-            transient_error_wait_time * 2, TRANSIENT_ERROR_WAIT_TIME_MAX
-        )
+        time.sleep(timeout)
     # Handle rate limiting
     elif (
         error.response.status_code == 403
         and "X-RateLimit-Remaining" in error.response.headers
         and int(error.response.headers["X-RateLimit-Remaining"]) == 0
     ):
-        timeout = int(error.response.headers["X-RateLimit-Reset"]) - time.time()
+        try:
+            now = calendar.timegm(
+                time.strptime(
+                    error.response.headers["Date"], "%a, %d %b %Y %H:%M:%S %Z"
+                )
+            )
+        except:
```

Comment:
To please the CI, please use `except ValueError` - also on second thought, this should probably log a warning so that we notice this eventually.

Reply:
makes sense, will do